File: app/cms/models/article.go
package models

import (
	"baize/app/common/baize/baizeUnix"
	"baize/app/common/commonModels"
)

type ArticleDQL struct {
	Title    string `form:"title" db:"title"`
	Category string `form:"category" db:"category"`
	Status   string `form:"status" db:"status"`
	Keyword  string `form:"keyword" db:"keyword"`
	commonModels.BaseEntityDQL
}

type ArticleDML struct {
	ArticleId   int64  `json:"articleId,string" db:"article_id"`
	Title       string `json:"title" db:"title" binding:"required"`
	Slug        string `json:"slug" db:"slug"`
	Summary     string `json:"summary" db:"summary"`
	Category    string `json:"category" db:"category" binding:"required"`
	CoverUrl    string `json:"coverUrl" db:"cover_url"`
	Content     string `json:"content" db:"content"`
	Status      string `json:"status" db:"status"`
	Sort        int64  `json:"sort" db:"sort"`
	PublishTime string `json:"publishTime" db:"publish_time"`
	CreateBy    string `json:"createBy" db:"create_by"`
	UpdateBy    string `json:"updateBy" db:"update_by"`
}

type ArticleVo struct {
	ArticleId   int64                `json:"articleId,string" db:"article_id"`
	Title       string               `json:"title" db:"title"`
	Slug        string               `json:"slug" db:"slug"`
	Summary     string               `json:"summary" db:"summary"`
	Category    string               `json:"category" db:"category"`
	CoverUrl    string               `json:"coverUrl" db:"cover_url"`
	Content     string               `json:"content" db:"content"`
	Status      string               `json:"status" db:"status"`
	Sort        int64                `json:"sort" db:"sort"`
	PublishTime string               `json:"publishTime" db:"publish_time"`
	CreateBy    string               `json:"createBy" db:"create_by"`
	CreateTime  *baizeUnix.BaiZeTime `json:"createTime" db:"create_time"`
	UpdateBy    string               `json:"updateBy" db:"update_by"`
	UpdateTime  *baizeUnix.BaiZeTime `json:"updateTime" db:"update_time"`
}

File: portal-ui/src/api/portal/article.ts
export interface ArticleItem {
  articleId?: string
  title: string
  slug: string
  summary: string
  category: string
  coverUrl?: string
  content: string
  status?: string
  sort?: number
  publishTime?: string
  createBy?: string
  updateBy?: string
}

export async function listArticles(query: Record<string, unknown> = {}) {
  const params = new URLSearchParams()
  Object.entries({ pageNum: 1, pageSize: 20, ...query }).forEach(([key, value]) => {
    const text = value == null ? '' : String(value)
    if (text !== '') {
      params.set(key, text)
    }
  })
  const response = await fetch(`/portal/articles?${params.toString()}`)
  const data = await response.json()
  if (!response.ok || String(data.code) !== '200') {
    throw new Error(data.msg || '文章加载失败')
  }
  return (data.rows || []) as ArticleItem[]
}

export async function getArticleBySlug(slug: string) {
  const response = await fetch(`/portal/articles/${encodeURIComponent(slug)}`)
  const data = await response.json()
  if (!response.ok || String(data.code) !== '200') {
    throw new Error(data.msg || '文章加载失败')
  }
  return data.data as ArticleItem
}
